### ISSUE TITLE
Scalar arguments should really be T rather than &[T]

### ISSUE THREAD

**Issue by Rufflewind** [stainless-steel/blas#26]
Some of the arguments should really be &T rather than &[T]:

```
pub unsafe fn zgemm(
    ...
    alpha: &[c64], 
    ...
    beta: &[c64], 
    ...
)
```

**Comment by IvanUkhov** [comment]
Ohh, gosh, how did that happen and has stayed under the radar all this time? I guess not many people use the C interface. Anyway, thanks for letting me know! Really appreciate it!

I’ll fix alphas and betas—and they should actually be without any reference, just T—but I guess there are more mistreated scalar parameters. Please feel free to report other bugs of this kind.

**Comment by IvanUkhov** [comment]
@Rufflewind, by the way, since you’re working with complex numbers, read carefully #22.

**Comment by IvanUkhov** [comment]
OK, I see now that this alpha-beta bug concerns only the complex functions. The reason is that the corresponding complex C functions happen to take those scalars by reference while their real counterparts take them by value. Apparently, I looked only at the real C functions, observed that they were taking scalars by value, and concluded that there was no need for any special treatment of function arguments like the one used in the Fortran interface where everything is by reference.

**Comment by IvanUkhov** [comment]
I’ve published a new version, v0.17, with alphas and betas fixed.

**Comment by Rufflewind** [comment]
Thanks for the quick fix!
